oak_remote_attestation/proto/v1/evidence.proto

```diff
@@ -0,0 +1,119 @@
+// a chain, where each element (called “layer”) represents a piece of software loaded into the VMs
+// memory. These layers are loaded sequentially, i.e. the previous layer loads the next layer. The
+// previous layer is also responsible for measuring the next layer, generating a private key and a
+// certificate for it.
+//
+// <https://trustedcomputinggroup.org/wp-content/uploads/TCG_DICE_Attestation_Architecture_r22_02dec2020.pdf>
+// <https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf>
+
+enum TeePlatform {
+  UNSPECIFIED = 0;
+  AMD_SEV_SNP = 1;
+  INTEL_TDX = 2;
+}
+
+message RootLayerEvidence {
+  // The platform providing the attestation report.
+  TeePlatform platform = 0;
+  // TEE-specific attestation report acting as a non-standard certificate for the
+  // Layer0 ECA public key.
+  bytes remote_attestation_report = 1;
+  // Serialised ECA public key for Layer0.
+  bytes eca_public_key = 2;
+}
+
+message LayerEvidence {
+  string layer_name = 0;
+  // CBOR/COSE/CWT ECA certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  bytes eca_certificate = 1;
+}
+
+message ApplicationKeys {
+  // CBOR/COSE/CWT encryption public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes encryption_public_key_certificate = 0;
+  // CBOR/COSE/CWT signing public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes signing_public_key_certificate = 1;
+}
+
+message Evidence {
+  // Layer0 attestation evidence.
+  RootLayerEvidence root_layer_evidence = 0;
+  // Layer1..LayerN-1 attestation evidence.
+  repeated LayerEvidence layer_evidence = 1;
+  // Final layer keys.
+  optional ApplicationKeys application_keys = 2;
+}
+
+// Endorsement contains statements that some entity (e.g., a hardware provider) vouches for the
+// integrity of claims about the TEE or the software running on it.
+// The name is chosen to match the RATS terminology:
+// <https://www.rfc-editor.org/rfc/rfc9334.html#name-endorsements>
+message Endorsement {
+  // The serialized TEE certificate(s). The details of the format and how the certificate(s) are
+  // encoded into this byte array are implementation-specific. In case of AMD-SEV-SNP, as described
+  // in https://www.amd.com/system/files/TechDocs/57230.pdf, there are three different certificates
+  // packaged in two different files.
+  bytes tee_certificates = 1;
+
+  // Transparent Release endorsement.
+  TransparentReleaseEndorsement transparent_release_endorsement = 2;
+}
+
+message TransparentReleaseEndorsement {
+  // The serialized binary endorsement statement from the transparent release process.
+  bytes endorsement_statement = 1;
+  // The serialized Rekor LogEntry as proof of the inclusion of the endorsement statement in Rekor.
+  bytes rekor_log_entry = 2;
+  // Public key of rekor at the time of uploading the endorsement statement. This is required for
```

Comment:
Wait, why is the Rekor public key here? Couldn't the server just put whatever key it wants here? What's the point of providing a signature as well as the key to verify it against?

Reply:
Currently Rekor uses a long-lived signing key for singing the uploaded entries. But this is not a good practice and in future they will change it to a short-lived singing key. In that scenario, the public keys will have to be verified using a parent key of a sort. The proto here is designed with that scenario in mind. Only the server has the public key of Rekor at the time of uploading the entry, and has to send it to the client. The client uses that to verify the log entry, then uses the parent public key that it trusts to verify the Rekor public key that it has received from the server. 

In the current implementation, the client simply checks that the public key it has received from the server is the same as the public key that it knows and trusts.